src/util/srp.cpp

```diff
@@ -422,8 +422,13 @@ static SRP_Result H_nn(
 }
 
 static SRP_Result H_ns(mpz_t result, SRP_HashAlgorithm alg, const unsigned char *n,
-	size_t len_n, const unsigned char *bytes, uint32_t len_bytes)
+	size_t len_n, const unsigned char *bytes, size_t len_bytes)
 {
+#ifdef __GNUC__
+	// Silence warning
+	if (len_bytes >= PTRDIFF_MAX)
```

Comment:
I think changing this line would silence the warning as well:
https://github.com/minetest/minetest/blob/22b3a5585e48e0822eb97b1b9437b9af80763320/src/util/srp.cpp#L357

Reply:
Ah, that seems to be the real source of the issue.  I'll change it to 0.